WalletWasabi/WabiSabi/Models/RoundState.cs

```diff
@@ -9,29 +9,39 @@
 namespace WalletWasabi.WabiSabi.Models
 {
 	public record RoundState(
-		uint256 Id,
 		uint256? BlameOf,
 		CredentialIssuerParameters AmountCredentialIssuerParameters,
 		CredentialIssuerParameters VsizeCredentialIssuerParameters,
 		FeeRate FeeRate,
 		Phase Phase,
 		bool WasTransactionBroadcast,
+		TimeSpan InputRegistrationTimeout,
 		TimeSpan ConnectionConfirmationTimeout,
+		TimeSpan OutputRegistrationTimeout,
+		TimeSpan TransactionSigningTimeout,
+		long MinRegistrableAmount,
 		long MaxRegistrableAmount,
 		long MaxRegistrableVsize,
 		long MaxVsizeAllocationPerAlice,
 		MultipartyTransactionState CoinjoinState)
 	{
+		private uint256 _id;
+
+		public uint256 Id => _id ??= CalculateHash();
```

Comment:
`_id` is not nullable, so won't this leave it with the default value of 0?

Reply:
I finally remember. The problem is that there are tests that modify the `MaxVsizeAllocationPerAlice` value after the round is created and that changes the hash. It is hard to solve.

Reply:
I still don't understand how this works at all... Isn't the right hand side of ??= deadcode if `_id` is a `uint256` and not a `uint256?` ?

Reply:
What is called _nullable types_ doesn't exist in reality, there are no nullable and non-nullable types, all reference types are nullable.

When you turn on nullable types you are turning on an special analyzer that checkd if the value is null. That verification is done because a non-nullable type can contain null, otherwise it would be unnecessary.

Reply:
ah, i see, i mistakenly thought `uint256` was a struct, not a class, and therefore a value type (like `Scalar`)